app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -363,12 +390,59 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
         }
     }
 
-    private fun updateAccountProfile(profile: Profile) {
+    private fun updateAccountProfile(profile: Profile, error: Boolean = false) {
         launch {
             val preferenceFirefoxAccount =
-                findPreference<Preference>(context!!.getPreferenceKey(pref_key_account))
-            preferenceFirefoxAccount?.title = profile.displayName.orEmpty()
-            preferenceFirefoxAccount?.summary = profile.email.orEmpty()
+                findPreference<AccountPreference>(context!!.getPreferenceKey(pref_key_account))
```

Comment:
`context!!` may fail if `SettingsFragment` isn't attached by the time this coroutine runs.